tokio/src/fs/file.rs

```diff
@@ -52,12 +52,24 @@ use std::fs::File as StdFile;
 /// Reading and writing to a `File` is usually done using the convenience
 /// methods found on the [`AsyncReadExt`] and [`AsyncWriteExt`] traits.
 ///
+/// # Warning
+/// [`std::fs::File`][std] is often used to represent a generic Posix file.
+/// You might not want to do that here. This file implementation is specialized
+/// for filesystem access, which does not have a concept of readiness, and so does
+/// not use the normal pattern of nonblocking IO operations and epoll (instead
+/// relying on the blocking thread pool). You should not attempt to do this for
+/// files which support nonblocking IO and polling, such as tun devices.
```

Comment:
I would probably turn this sentence around and start by saying that this type is specialized for actual files on disk, and therefore it should not be used for special file devices that support nonblocking IO.